Add `@Nullable` annotations or initialize nonnull fields

arteam opened the pull request:
We now run a static analyzer during our build (#2204) which helps us to catch NPE during compilation. To make it work, we should annotate actual nullable fields with the `@Nullable` annotation, otherwise the analyzer will always consider such fields as non-null by default and will report violations when a null value is assigned to them.

jplock commented:
If we uncomment the line I commented in the pom, we can have Travis enforce this too. Do you think we should do that?

arteam commented:
Yes, I went through all null check violation reports (even in the tests) and fixed them. I've changed the reporting level to ERROR. Now the build will stop if the compiler finds a violation.

jplock commented on dropwizard-benchmarks/pom.xml:
```diff
@@ -23,6 +23,21 @@
     <name>Dropwizard Benchmarks</name>
 
+    <build>
+        <plugins>
```
The nullaway configuration also allows you to exclude packages there too if we want to excludes these packages in the main pom

arteam replied:
Yes, it will be probably a better approach.
